File: scripts/batch_update_module1.py
#!/usr/bin/env python3
"""
Batch update script for Module 1 HTML files
Applies the new template structure to all remaining files
"""

import os
import re

# Template for the updated HTML structure
HTML_TEMPLATE = """<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    
    <!-- SEO Meta Tags -->
    <title>{title} - PHP WordPress Course</title>
    <meta name="description" content="{description}">
    <meta name="keywords" content="{keywords}">
    <meta name="author" content="PHP WordPress Course">
    
    <!-- Favicon -->
    <link rel="icon" type="image/png" href="/favicon.png">
    <link rel="apple-touch-icon" href="/favicon.png">
    
    <!-- CSS -->
    <link rel="stylesheet" href="/assets/css/main.css">
</head>
<body>
    <!-- Skip to main content -->
    <a href="#main-content" class="sr-only">Skip to main content</a>
    
    <div class="page-wrapper">
        {header}
        {progress_bar}
        {breadcrumb}
        
        <!-- Main Content -->
        <main id="main-content" class="main-content" role="main">
            <div class="container">
                <div class="content-with-sidebar">
                    {sidebar}
                    
                    <!-- Main Lesson Content -->
                    <article class="lesson-content">
                        <header class="lesson-header">
                            <h1>{title}</h1>
                            <div class="lesson-meta">
                                <div class="lesson-meta-item">
                                    <svg width="20" height="20" fill="currentColor">
                                        <path d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"/>
                                    </svg>
                                    <span>Duration: {duration}</span>
                                </div>
                                <div class="lesson-meta-item">
                                    <svg width="20" height="20" fill="currentColor">
                                        <path d="M12 14l9-5-9-5-9 5 9 5z"/>
                                    </svg>
                                    <span>Module 1, {session}</span>
                                </div>
                            </div>
                        </header>

                        {learning_objectives}
                        
                        <!-- Lesson Body -->
                        <div class="lesson-body">
                            {content}
                            
                            {homework}
                            
                            {resources}
                        </div>

                        {lesson_navigation}
                    </article>
                </div>
            </div>
        </main>

        {footer}
    </div>

    <!-- Back to Top -->
    <button id="back-to-top" class="back-to-top" aria-label="Back to top">
        <svg width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2">
            <path d="M12 19V5M12 5l-7 7M12 5l7 7"/>
        </svg>
    </button>

    <!-- JavaScript -->
    <script src="/assets/js/navigation.js"></script>
    <script src="/assets/js/site-config.js"></script>
</body>
</html>"""

# Common header HTML
HEADER_HTML = """        <!-- Header -->
        <header class="site-header" role="banner">
            <div class="header-container">
                <div class="site-branding">
                    <a href="/" class="site-logo">
                        <h1 class="site-title">PHP WordPress Development</h1>
                    </a>
                </div>
                
                <nav class="main-navigation" role="navigation" aria-label="Main navigation">
                    <button class="mobile-menu-btn" aria-label="Toggle navigation" aria-expanded="false">
                        <span></span>
                        <span></span>
                        <span></span>
                    </button>
                    
                    <div class="nav-menu">
                        <ul class="nav-list">
                            <li class="nav-item"><a href="/" class="nav-link">Home</a></li>
                            <li class="nav-item dropdown">
                                <button class="nav-link dropdown-toggle active" aria-haspopup="true">Modules</button>
                                <div class="dropdown-menu">
                                    <a href="/module1.html" class="dropdown-item active">Module 1: Web Fundamentals</a>
                                    <a href="/module2.html" class="dropdown-item">Module 2: PHP Fundamentals</a>
                                    <a href="/module3.html" class="dropdown-item">Module 3: MySQL Database</a>
                                    <a href="/module4.html" class="dropdown-item">Module 4: WordPress & Docker</a>
                                    <a href="/module5.html" class="dropdown-item">Module 5: Theme Development</a>
                                    <a href="/module6.html" class="dropdown-item">Module 6: Plugin Development</a>
                                    <a href="/module7.html" class="dropdown-item">Module 7: Advanced WordPress</a>
                                    <a href="/module8.html" class="dropdown-item">Module 8: Deployment</a>
                                    <a href="/module9.html" class="dropdown-item">Module 9: Final Project</a>
                                </div>
                            </li>
                            <li class="nav-item"><a href="/resources.html" class="nav-link">Resources</a></li>
                            <li class="nav-item"><a href="/about.html" class="nav-link">About</a></li>
                        </ul>
                    </div>
                </nav>
                
                <div class="search-container">
                    <div class="search-input-wrapper">
                        <svg class="search-icon" width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                            <path fill-rule="evenodd" d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z" clip-rule="evenodd" />
                        </svg>
                        <input type="search" class="search-input" placeholder="Search lessons..." aria-label="Search">
                    </div>
                    <div class="search-results"></div>
                </div>
            </div>
        </header>"""

# Progress bar HTML
PROGRESS_BAR_HTML = """        <!-- Progress Bar -->
        <div class="progress-container">
            <div class="progress-header">
                <h2 class="progress-title">Course Progress</h2>
                <span class="progress-text">Loading...</span>
            </div>
            <div class="progress-bar">
                <div class="progress-bar-fill">
                    <span class="progress-bar-text"></span>
                </div>
            </div>
        </div>"""

# Common footer HTML
FOOTER_HTML = """        <!-- Footer -->
        <footer class="site-footer" role="contentinfo">
            <div class="footer-container">
                <div class="footer-content">
                    <div class="footer-section footer-about">
                        <h3>PHP WordPress Development</h3>
                        <p>Complete Web Development Course</p>
                    </div>
                    
                    <div class="footer-section">
                        <h4>Quick Links</h4>
                        <ul class="footer-links">
                            <li><a href="/">Home</a></li>
                            <li><a href="/module1.html">Module 1</a></li>
                            <li><a href="/resources.html">Resources</a></li>
                        </ul>
                    </div>
                    
                    <div class="footer-section">
                        <h4>Current Module</h4>
                        <ul class="footer-links">
                            {footer_links}
                        </ul>
                    </div>
                    
                    <div class="footer-section">
                        <h4>Support</h4>
                        <ul class="footer-links">
                            <li><a href="/help.html">Help Center</a></li>
                            <li><a href="/faq.html">FAQ</a></li>
                            <li><a href="/contact.html">Contact</a></li>
                        </ul>
                    </div>
                </div>
                
                <div class="footer-bottom">
                    <div class="footer-bottom-content">
                        <p class="copyright">&copy; 2025 PHP WordPress Development Course</p>
                        <nav class="footer-bottom-links">
                            <a href="/privacy.html">Privacy</a>
                            <span class="separator">|</span>
                            <a href="/terms.html">Terms</a>
                        </nav>
                    </div>
                </div>
            </div>
        </footer>"""

# Module 1 structure
module1_structure = {
    "session3": {
        "title": "CSS Basics",
        "files": [
            ("introduction_to_css.html", "Introduction to CSS and Its Purpose", "45 minutes"),
            ("css_syntax_and_selectors.html", "CSS Syntax and Selectors", "45 minutes"),
            ("css_implementation_methods.html", "Inline, Internal, and External CSS", "30 minutes"),
            ("css_colors_fonts_text.html", "Working with Colors, Fonts, and Text", "45 minutes"),
            ("css_box_model.html", "The CSS Box Model", "45 minutes"),
            ("homework_css_profile.html", "Homework: Style Your Profile Page", "60 minutes")
        ]
    },
    "session6": {
        "title": "JavaScript Fundamentals",
        "files": [
            ("js_intro.html", "Introduction to JavaScript", "45 minutes"),
            ("js_syntax_fundamentals.html", "JavaScript Syntax and Variables", "45 minutes"),
            ("js_operators_and_expressions.html", "Operators and Expressions", "30 minutes"),
            ("js_control_flow.html", "Control Flow", "45 minutes"),
            ("js_functions_and_scope.html", "Functions and Scope", "45 minutes"),
            ("homework_simple_js_programs.html", "Homework: Simple JS Programs", "60 minutes")
        ]
    }
}

def create_breadcrumb(title):
    """Generate breadcrumb HTML"""
    return f"""        <!-- Breadcrumb -->
        <nav class="breadcrumb container" aria-label="Breadcrumb">
            <ol class="breadcrumb-list">
                <li class="breadcrumb-item">
                    <a href="/">Home</a>
                    <span class="breadcrumb-separator">/</span>
                </li>
                <li class="breadcrumb-item">
                    <a href="/module1.html">Module 1</a>
                    <span class="breadcrumb-separator">/</span>
                </li>
                <li class="breadcrumb-item">
                    <span aria-current="page">{title}</span>
                </li>
            </ol>
        </nav>"""

def create_sidebar(session_title, files, current_index):
    """Generate sidebar HTML"""
    sidebar = f"""                    <!-- Sidebar -->
                    <aside class="sidebar">
                        <div class="sidebar-nav">
                            <h3 class="sidebar-title">Module 1: {session_title}</h3>
                            <div class="sidebar-section">
                                <h4 class="sidebar-section-title">Lessons</h4>"""
    
    for i, (filename, title, _) in enumerate(files):
        active = " active" if i == current_index else ""
        sidebar += f"""
                                <a href="/01module/{filename}" class="sidebar-link{active}">{title}</a>"""
    
    sidebar += """
                            </div>
                        </div>
                    </aside>"""
    
    return sidebar

def create_lesson_navigation(files, current_index):
    """Generate lesson navigation HTML"""
    nav_html = """                        <!-- Lesson Navigation -->
                        <div class="lesson-navigation">"""
    
    # Previous button
    if current_index > 0:
        prev_file, prev_title, _ = files[current_index - 1]
        nav_html += f"""
                            <a href="/01module/{prev_file}" class="lesson-nav-button prev">
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"/>
                                </svg>
                                <span>
                                    <small>Previous</small><br>
                                    {prev_title}
                                </span>
                            </a>"""
    else:
        nav_html += """
                            <a href="/module1.html" class="lesson-nav-button prev">
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"/>
                                </svg>
                                <span>
                                    <small>Back to</small><br>
                                    Module 1 Overview
                                </span>
                            </a>"""
    
    nav_html += """
                            
                            <button class="complete-lesson-btn">
                                Mark as Complete
                            </button>"""
    
    # Next button
    if current_index < len(files) - 1:
        next_file, next_title, _ = files[current_index + 1]
        nav_html += f"""
                            
                            <a href="/01module/{next_file}" class="lesson-nav-button next">
                                <span>
                                    <small>Next Lesson</small><br>
                                    {next_title}
                                </span>
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"/>
                                </svg>
                            </a>"""
    
    nav_html += """
                        </div>"""
    
    return nav_html

def create_learning_objectives(title):
    """Generate learning objectives based on title"""
    objectives_html = """                        <!-- Learning Objectives -->
                        <div class="lesson-objectives">
                            <h2>Learning Objectives</h2>
                            <ul>"""
    
    # Add appropriate objectives based on title
    if "CSS" in title:
        objectives_html += """
                                <li>Understand CSS fundamentals and syntax</li>
                                <li>Learn how to style HTML elements</li>
                                <li>Apply CSS to create visually appealing pages</li>
                                <li>Master CSS selectors and properties</li>"""
    elif "JavaScript" in title or "JS" in title:
        objectives_html += """
                                <li>Understand JavaScript basics</li>
                                <li>Learn programming fundamentals</li>
                                <li>Write interactive web applications</li>
                                <li>Master JavaScript syntax and concepts</li>"""
    else:
        objectives_html += """
                                <li>Master the concepts covered in this lesson</li>
                                <li>Apply knowledge to practical examples</li>
                                <li>Build confidence with hands-on practice</li>
                                <li>Prepare for more advanced topics</li>"""
    
    objectives_html += """
                            </ul>
                        </div>"""
    
    return objectives_html

# Print file update list
print("Files to update:")
for session_key, session_data in module1_structure.items():
    print(f"\n{session_data['title']}:")
    for filename, title, duration in session_data['files']:
        print(f"  - {filename}: {title}")

print("\n✅ Batch update script ready!")
print("This script will help update the remaining Module 1 files with the new template structure.")

File: scripts/batch-update-module1.js
#!/usr/bin/env node

/**
 * Batch Update Script for Module 1 HTML Files
 * This script updates all Module 1 lesson files to use the new template structure
 */

const fs = require('fs');
const path = require('path');

// Module 1 lesson structure
const module1Structure = {
    session1: {
        title: "Introduction to Web Development & Setup",
        lessons: [
            { file: "course_introduction.html", title: "Course Introduction and Expectations", duration: "30 minutes", completed: true },
            { file: "how_web_works.html", title: "How the Web Works: Servers, Browsers, HTTP", duration: "45 minutes", completed: true },
            { file: "development_environment.html", title: "Setting Up Your Development Environment", duration: "60 minutes", completed: true },
            { file: "first_html_page.html", title: "Creating Your First HTML Page", duration: "45 minutes", completed: true },
            { file: "homework_setup.html", title: "Homework: Setup GitHub and Docker", duration: "30 minutes" }
        ]
    },
    session2: {
        title: "HTML Fundamentals", 
        lessons: [
            { file: "html_structure_syntax.html", title: "HTML Document Structure and Syntax", duration: "45 minutes", completed: true },
            { file: "essential_html_tags.html", title: "Essential HTML Tags", duration: "60 minutes" },
            { file: "html_forms_inputs.html", title: "HTML Forms and Inputs", duration: "45 minutes" },
            { file: "html_validation_practices.html", title: "HTML Validation and Best Practices", duration: "30 minutes" },
            { file: "homework_html_profile.html", title: "Homework: Create a Personal Profile Page", duration: "60 minutes" }
        ]
    },
    session3: {
        title: "CSS Basics",
        lessons: [
            { file: "introduction_to_css.html", title: "Introduction to CSS and Its Purpose", duration: "45 minutes" },
            { file: "css_syntax_and_selectors.html", title: "CSS Syntax and Selectors", duration: "45 minutes" },
            { file: "css_implementation_methods.html", title: "Inline, Internal, and External CSS", duration: "30 minutes" },
            { file: "css_colors_fonts_text.html", title: "Working with Colors, Fonts, and Text", duration: "45 minutes" },
            { file: "css_box_model.html", title: "The CSS Box Model", duration: "45 minutes" },
            { file: "homework_css_profile.html", title: "Homework: Style Your Profile Page", duration: "60 minutes" }
        ]
    },
    session4: {
        title: "CSS Layout & Responsive Design",
        lessons: [
            { file: "css_layout_tech.html", title: "CSS Layout Techniques", duration: "60 minutes" },
            { file: "responsive_design.html", title: "Introduction to Responsive Design", duration: "45 minutes" },
            { file: "media_queries.html", title: "Media Queries", duration: "45 minutes" },
            { file: "mobile_first.html", title: "Mobile-First Approach", duration: "30 minutes" },
            { file: "homework_responsive_profile.html", title: "Homework: Make Your Profile Responsive", duration: "60 minutes" }
        ]
    },
    session5: {
        title: "CSS Frameworks & Best Practices",
        lessons: [
            { file: "bootstrap.html", title: "Introduction to Bootstrap", duration: "45 minutes" },
            { file: "bootstrap_grid.html", title: "Using Bootstrap Grid System", duration: "45 minutes" },
            { file: "bootstrap_components.html", title: "Bootstrap Components", duration: "45 minutes" },
            { file: "bootstrap_utilities.html", title: "Bootstrap Utilities", duration: "30 minutes" },
            { file: "css_organization_best_practices.html", title: "CSS Organization and Best Practices", duration: "30 minutes" },
            { file: "css_preprocessors.html", title: "CSS Preprocessors Overview (SASS/SCSS)", duration: "30 minutes" },
            { file: "homework_bootstrap_profile.html", title: "Homework: Recreate Profile with Bootstrap", duration: "60 minutes" }
        ]
    },
    session6: {
        title: "JavaScript Fundamentals",
        lessons: [
            { file: "js_intro.html", title: "Introduction to JavaScript", duration: "45 minutes" },
            { file: "js_syntax_fundamentals.html", title: "JavaScript Syntax, Variables, and Data Types", duration: "45 minutes" },
            { file: "js_operators_and_expressions.html", title: "Operators and Expressions", duration: "30 minutes" },
            { file: "js_control_flow.html", title: "Control Flow (Conditionals, Loops)", duration: "45 minutes" },
            { file: "js_functions_and_scope.html", title: "Functions and Scope", duration: "45 minutes" },
            { file: "homework_simple_js_programs.html", title: "Homework: Create Simple JavaScript Programs", duration: "60 minutes" }
        ]
    },
    session7: {
        title: "DOM Manipulation with JavaScript",
        lessons: [
            { file: "understanding_dom.html", title: "Understanding the Document Object Model (DOM)", duration: "45 minutes" },
            { file: "dom_selection_manipulation.html", title: "Selecting and Manipulating DOM Elements", duration: "45 minutes" },
            { file: "event_handling.html", title: "Event Handling", duration: "45 minutes" },
            { file: "creating_removing_elements.html", title: "Creating and Removing Elements", duration: "30 minutes" },
            { file: "form_validation.html", title: "Form Validation with JavaScript", duration: "45 minutes" },
            { file: "homework_interactive.html", title: "Homework: Add Interactivity to Your Page", duration: "60 minutes" }
        ]
    },
    session8: {
        title: "Modern JavaScript & jQuery",
        lessons: [
            { file: "es6_overview.html", title: "ES6+ Features Overview", duration: "45 minutes" },
            { file: "jquery_intro.html", title: "Introduction to jQuery", duration: "45 minutes" },
            { file: "jquery_dom_manipulation.html", title: "DOM Manipulation with jQuery", duration: "45 minutes" },
            { file: "jquery_animations_and_effects.html", title: "jQuery Animations and Effects", duration: "30 minutes" },
            { file: "jquery_ajax.html", title: "AJAX Basics with jQuery", duration: "45 minutes" },
            { file: "homework_jquery_refactor.html", title: "Homework: Refactor Your Code Using jQuery", duration: "60 minutes" }
        ]
    },
    session9: {
        title: "Introduction to PHP",
        lessons: [
            { file: "php_and_wordpress.html", title: "What is PHP and Why It's Important for WordPress", duration: "45 minutes" },
            { file: "php_setup_xampp_mamp.html", title: "Setting Up a Local Server (XAMPP/MAMP)", duration: "45 minutes" },
            { file: "php_syntax.html", title: "PHP Syntax and Basic Constructs", duration: "45 minutes" },
            { file: "php_variables_data_and_operators.html", title: "Variables, Data Types, and Operators", duration: "45 minutes" },
            { file: "php_includes.html", title: "Including Files and PHP in HTML", duration: "30 minutes" },
            { file: "homework_simple_php.html", title: "Homework: Create a Simple PHP Script", duration: "60 minutes" }
        ]
    },
    session10: {
        title: "Mini-Project: Static Website",
        lessons: [
            { file: "planning_website.html", title: "Planning a Multi-Page Website", duration: "45 minutes" },
            { file: "creating_layout.html", title: "Creating a Consistent Layout", duration: "60 minutes" },
            { file: "adding_interactivity_with_js.html", title: "Adding Interactivity with JavaScript", duration: "60 minutes" },
            { file: "php_header_footer.html", title: "Basic PHP Includes for Header and Footer", duration: "45 minutes" },
            { file: "project_static_site.html", title: "Final Project: 5-Page Static Website", duration: "180 minutes" }
        ]
    }
};

// Function to get navigation for a lesson
function getLessonNavigation(sessionKey, lessonIndex) {
    const sessions = Object.keys(module1Structure);
    const currentSession = module1Structure[sessionKey];
    const currentSessionIndex = sessions.indexOf(sessionKey);
    
    let prevLesson = null;
    let nextLesson = null;
    
    // Previous lesson
    if (lessonIndex > 0) {
        prevLesson = currentSession.lessons[lessonIndex - 1];
    } else if (currentSessionIndex > 0) {
        const prevSession = module1Structure[sessions[currentSessionIndex - 1]];
        prevLesson = prevSession.lessons[prevSession.lessons.length - 1];
    }
    
    // Next lesson 
    if (lessonIndex < currentSession.lessons.length - 1) {
        nextLesson = currentSession.lessons[lessonIndex + 1];
    } else if (currentSessionIndex < sessions.length - 1) {
        const nextSession = module1Structure[sessions[currentSessionIndex + 1]];
        nextLesson = nextSession.lessons[0];
    }
    
    return { prevLesson, nextLesson };
}

// Function to generate sidebar HTML
function generateSidebar(sessionKey, lessonIndex) {
    const session = module1Structure[sessionKey];
    const currentLesson = session.lessons[lessonIndex];
    
    let sidebarHTML = `
                    <!-- Sidebar -->
                    <aside class="sidebar">
                        <div class="sidebar-nav">
                            <h3 class="sidebar-title">Module 1: ${session.title}</h3>
                            <div class="sidebar-section">
                                <h4 class="sidebar-section-title">Lessons</h4>`;
    
    session.lessons.forEach((lesson, index) => {
        const isActive = index === lessonIndex ? ' active' : '';
        sidebarHTML += `
                                <a href="/01module/${lesson.file}" class="sidebar-link${isActive}">${lesson.title}</a>`;
    });
    
    sidebarHTML += `
                            </div>
                        </div>
                    </aside>`;
    
    return sidebarHTML;
}

// Function to generate lesson navigation HTML
function generateLessonNav(prevLesson, nextLesson) {
    let navHTML = `
                        <!-- Lesson Navigation -->
                        <div class="lesson-navigation">`;
    
    if (prevLesson) {
        navHTML += `
                            <a href="/01module/${prevLesson.file}" class="lesson-nav-button prev">
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"/>
                                </svg>
                                <span>
                                    <small>Previous</small><br>
                                    ${prevLesson.title}
                                </span>
                            </a>`;
    } else {
        navHTML += `
                            <a href="/module1.html" class="lesson-nav-button prev">
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"/>
                                </svg>
                                <span>
                                    <small>Back to</small><br>
                                    Module 1 Overview
                                </span>
                            </a>`;
    }
    
    navHTML += `
                            
                            <button class="complete-lesson-btn">
                                Mark as Complete
                            </button>`;
    
    if (nextLesson) {
        navHTML += `
                            
                            <a href="/01module/${nextLesson.file}" class="lesson-nav-button next">
                                <span>
                                    <small>Next Lesson</small><br>
                                    ${nextLesson.title}
                                </span>
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"/>
                                </svg>
                            </a>`;
    } else {
        navHTML += `
                            
                            <a href="/module2.html" class="lesson-nav-button next">
                                <span>
                                    <small>Next Module</small><br>
                                    Module 2: PHP Fundamentals
                                </span>
                                <svg width="20" height="20" viewBox="0 0 20 20" fill="currentColor">
                                    <path d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010-1.414l-4 4a1 1 0 01-1.414 0z"/>
                                </svg>
                            </a>`;
    }
    
    navHTML += `
                        </div>`;
    
    return navHTML;
}

// Export functions for use in the update process
module.exports = {
    module1Structure,
    getLessonNavigation,
    generateSidebar,
    generateLessonNav
};